### ISSUE TITLE
Does not prompt for missing variables when --var-file argument is given

### ISSUE THREAD

**Issue by chids** [hashicorp/terraform#6554]
### Terraform Version

```
$ terraform --version
Terraform v0.6.15
```
### Terraform Configuration Files

`main.tf`

``` hcl
variable "FOO" { type = "string" }
variable "BAR" { type = "string" }

resource "null_resource" "dummy-echo" {
  provisioner "local-exec" {
    command = "echo ${var.FOO} ${var.BAR}"
  }
}
```

`foo.tfvars`

``` hcl
FOO = "bar"
```
### Expected Behavior

Terraform should prompt for all missing variable values.
### Actual Behavior

Terraform does not prompt for the value of `BAR` if one supplies a `.tfvars` file even if that file does not define `BAR`.
### Steps to Reproduce

Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
   - 👉 prompts for `FOO` and `BAR`
2. Delete state files
3. `terraform apply -var-file=foo.tfvars`
   - 💥 `Required variable not set: BAR`
### References

Mentioned / reported by @jmehnle in #1230 and mentioned/referenced in #1899

**Comment by eedwardsdisco** [comment]
+1 I've encountered this as well on 0.6.15

**Comment by missingcharacter** [comment]
Same behavior happens after specifying `-input=true`
```
$ terraform --version
Terraform v0.6.16
...
$ terraform apply -input=true -var-file=foo.tfvars
Required variable not set: BAR
```

**Comment by rsmitty** [comment]
I too am seeing what I believe is a version of this behavior. When specifying a var-file, I get the following output:

```bash
$ terraform apply -var-file=coreos.tfvars
Error configuring: 3 error(s) occurred:

* 1:3: unknown variable accessed: var.numControllers in:

${var.numControllers}
```

Renaming the tfvars file to `terraform.tfvars` and running without the -var-file flag prompts for the required, unspecified variables as expected.

**Comment by mitchellh** [comment]
This has been fixed since 0.7, I ran your exact example on it and it works!
